### ISSUE TITLE
Doesn't seem to work?

### ISSUE THREAD

**Issue by unional** [alexisvincent/systemjs-config-builder#18]
I follow the example to do:

```sh
// in an empty folder.
yarn global add systemjs-tools
yarn add systemjs systemjs-nodelibs systemjs-plugin-babel react react-dom
systemjs config
```

But the `systemjs config` just exit and no `generated.config.js` is created.

**Comment by unional** [comment]
I'm using `yarn`: 0.19.1, `npm`: 4.0.5, `node`: 7.5.0

**Comment by alexisvincent** [comment]
I think the command has changed to generate-config. Run systemjs --help to verify.

**Comment by unional** [comment]
Thanks. It now complains:

```sh
:: exiting :: couldn't find a valid systemjs-tools config
```

**Comment by alexisvincent** [comment]
Damn. I recently changed how systemjs initializes itself. Just heading in to a meeting now. Can fix after that.

**Comment by unional** [comment]
No problem. Take your time. 🌷 
Thanks for the good work.

**Comment by alexisvincent** [comment]
@unional Just had a look, I actually use the config for `systemjs-tools`, so to fix this, in your package.json, add: 
```javascript
{
    ...
    "systemjs-tools": {}
    ...
}
```

This will indicate to config builder where the root is. You can also customize the cache file location via

```javascript
{
    ...
    "systemjs-tools": {
        "cache": ".myCustomeCacheFile.json"
    }
    ...
}
```

**Comment by alexisvincent** [comment]
@unional did this solve your issue?

**Comment by unional** [comment]
I see. Have not try it yet.
Is it possible to make it not rely on this?
IMO it sounds like an optional configuration.
Are you doing a lookup in folder structure to find the root `package.json`?

**Comment by alexisvincent** [comment]
Yes, and then it uses the file it finds to act as a 'root'. But can probably offer a default. Although this isn't high priority for me right now

**Comment by simoncode42** [comment]
Could you add some documentation on how the cache works?  Every time I run generate-config, it takes a long time to complete, and don't see any cache files created.

**Comment by simondiep** [comment]
Could you add some documentation on how the cache works? Every time I run generate-config, it takes a long time to complete, and don't see any cache files created.

**Comment by alexisvincent** [comment]
I don't have time to write docs right now, since I'm right in the middle of exams, but you can take a look here
https://github.com/alexisvincent/systemjs-tools/blob/master/lib/cli.js#L21

It gets the cache out of the systemjs cache. You should see a  `.systemjs.cache.json` file, which you configure by the cache key in the config. Checkout https://github.com/alexisvincent/systemjs-tools/blob/master/docs/config.md.

**Comment by simondiep** [comment]
There might be a bug with the cache generation, as I see a systemjs.cache file created but with contents of `undefined`.

https://github.com/alexisvincent/systemjs-config-builder/blob/master/lib/index.js#L425
